### PR TITLE
Arquillian porting package improvements

### PR DESCRIPTION
**Pull request by OndroMih**
* Separate runtime specific parts of Arquillian execution 
* Prepare support for EAR package deployment

### CODE REVIEW

File: pom.xml
Comment (scottkurz) [inline pom.xml:34]: Thanks

File: jakarta.batch.arquillian.exec.example/README.md
Comment (scottkurz) [inline jakarta.batch.arquillian.exec.example/README.md:17]: Assume this isn't actually possible with a publicly-available build still ?

Does the TCK test DB need to be configured?

File: jakarta.batch.arquillian.exec.example/README.md
Comment (scottkurz) [inline jakarta.batch.arquillian.exec.example/README.md:2]: I think the Jakarta process involves a careful review of the copyright.  I know ideally we do this before merging but if it's just me and you involved for now we can I'm sure resolve it before final release, if necessary.

File: jakarta.batch.arquillian.exec.example/test.properties
Comment (scottkurz) [inline jakarta.batch.arquillian.exec.example/test.properties:1]: Before we had test.properties in .gitignore and were copying from jakarta.batch.arquillian.exec/tck.default.sleep.time.properties

Now we're just committing the file though we still have some .gitignore references.   I guess we could duplicate the file though it'd be better to copy at execution time. 

Either way itt'd be nice to fix this before merging.

**Review by scottkurz — CHANGES_REQUESTED** [review]
## 1
Re:  module naming - All the '*.exec' projects are example executions... 
so I might rename:
    jakarta.batch.arquillian.exec.example =>
    jakarta.batch.arquillian.exec.example
and rename:
    jakarta.batch.arquillian.exec =>
    jakarta.batch.arquillian.exec-parent  (or      jakarta.batch.arquillian.parent or something like that).

## 2 
 I refactored a couple things btw. parent/child and did a PR to your PR branch:  https://github.com/OndroMih/batch-tck/pull/6    Please see what you think.  Hope you don't mind getting input this way.

File: jakarta.batch.arquillian.exec.example/README.md
Comment (OndroMih) [reply jakarta.batch.arquillian.exec.example/README.md:2]: I'm not sure whether you suggest a change here or we can leave this as is for this PR. I followed the [Eclipse Project Handbook](https://www.eclipse.org/projects/handbook/#ip-copyright-headers), which states that "Alternatively, project teams can adopt a more generic copyright header for each file and include the identities of the copyright holders in the notice file."

File: jakarta.batch.arquillian.exec.example/test.properties
Comment (OndroMih) [reply jakarta.batch.arquillian.exec.example/test.properties:1]: I committed the test.properties file into this project because it's only an example. It shouldn't be used as a maven artifact, people should only use it as a template for their TCK execution project. In that case I believe it makes more sense if people just commit their test.properties file directly rather than generate it at execution time from a template.

The best solution would be if the maven project could download the template from some maven artifact and then copy it to test.properties. But there's no such maven artifact. All projects that contain a file that could be used as a template are plain POM artifacts, which don't contain it in the deployed maven artifact. 

Maybe we should create a template test.properties file in the `com.ibm.jbatch.tck` artifact so that it can be copied from it? It could also be used in the `com.ibm.jbatch.tck.exec` RI execution project, which now contains its own copy.

If you agree, I'll add the template test.properties file there and modify both RI and Arquillian exec projects to copy it from the maven dependency.

File: jakarta.batch.arquillian.exec.example/README.md
Comment (OndroMih) [reply jakarta.batch.arquillian.exec.example/README.md:17]: Now, the public GF build doesn't pass the TCK. We'll have to wait until GF team adds support for the current spec requirements.

**Review by scottkurz — APPROVED** [review]

File: jakarta.batch.arquillian.exec.example/test.properties
Comment (OndroMih) [reply jakarta.batch.arquillian.exec.example/test.properties]: OK. So I added the `tck.default.sleep.time.properties` template file into the `com.ibm.jbatch.tck` artifact and adapted  arquillian exec parent to copy that file from the `com.ibm.jbatch.tck` artifact to test.properties if that file doesn't exist yet. The file `test.properties` remains ignored by git so the old semantics remains - the test.properties file is generated based on a template and can be changed later for local execution, without committing it to git. 

Projects based on the Arquillian parent created by those who want to execute the TCK against the implementation of their choice can add the generated `test.properties` file to git as their project will be in a different repository.

File: jakarta.batch.arquillian.exec.example/.gitignore
Comment (OndroMih) [reply jakarta.batch.arquillian.exec.example/.gitignore]: I deleted the `test.properties` file, it's now generated from the `com.ibm.jbatch.tck` artifact.

**Comment by OndroMih** [comment]
Hi @scottkurz :

## Ad. 1

I've renamed the Arquillian example projects as follows:

* `jakarta.batch.arquillian.exec-parent` - parent project that can be used as a parent for custom TCK execution projects using an Arquillian container
* `jakarta.batch.arquillian.exec` - an example execution project that shows how to use the parent project. Uses `exec-parent` as the parent and shows how to configure GlassFish and OpenLiberty. GlassFish still doesn't pass the TCK, once it will integrate the latest JBatch version, I will tune the example configuration.

**Comment by scottkurz** [comment]
Taking some vacation so not sure if I'll get back to this before New Year.  Thanks.

**Comment by scottkurz** [comment]
Hi @OndroMih ..just getting back to this.   It looks good, thanks for doing this.

Two comments:
1.  I made another "code review PR": https://github.com/OndroMih/batch-tck/pull/7 to your PR branch like before rather than review comments.   These were just things I thought could be clearer.  Not strictly related but maybe you don't mind pulling in.
2.  I see you removed the 'antrun' execution that printed the warning message.   I thought it was mildly helpful but I'm OK removing it if the README itself is enough doc.

**Comment by scottkurz** [comment]
@OndroMih  not sure if you saw this thread either:  https://www.eclipse.org/lists/jakartaee-platform-dev/msg03025.html but it seems to suggest we use:  **jakarta.tck.batch.*** as our pkg prefix for new code.  (I didn't read the whole thread I admit, however).

**Comment by rmannibucau** [comment]
Wait since jakarta.* is forbidden for app code (so part of tck simulating an app).

**Comment by OndroMih** [comment]
I noticed that there has been some discussion about the package names (https://www.eclipse.org/lists/jakartaee-platform-dev/msg03134.html). There are a few options that were suggested and a quick poll, which shows that **ee.jakarta.tck.batch** is the most popular pattern. I'd go with that, as discussed, Eclipse now owns the jakarta.ee domain so it shouldn't be a problem with ee.jakarta prefix and the rest isn't important for us. Even if the platform decides on something else, it looks like it won't be mandatory, as many people commented here:

**Comment by OndroMih** [comment]
I've synchronized my changes with the current master and renamed the packages to use the ee.jakarta.tck prefix. I left the module names to use the current names (jakarta.batch...), these names shouldn't cause issues. We can change them separately later if we want.

I'm going to work now on a separate PR for introducing support for various vehicles. Let's merge this PR so that I can create a another one on top of this one. I don't want to add everything together in a single PR because the vehicle support will touch a lot of files.

**Comment by scottkurz** [comment]
@OndroMih , looks good and thank you for all this contribution.

Knowing that you are trying to merge this in and do the follow-up, let me make some comments so we can do just that.

1. There was a minor merge error, I fixed with another PR to your branch: https://github.com/OndroMih/batch-tck/pull/8
2. I realized the **test.properties** aren't really relevant to the Arquillian execution, at least for Open Liberty.  (I don't know enough about Arquillian to be clear if other adapters could be different).   But they'd only apply to the failsafe JVM not to the separate managed server.    I think this is OK. The defaults have a good chance of working.  There are other ways to set these properties even if we haven't solved this problem nicely as we were trying to do.
3. I noticed some of the new parts have a LICENSE statement but not one including the copyright year and the owner at the top, only a reference to the NOTICE.   Was this intentional?   (Also I opened:  https://github.com/eclipse-ee4j/batch-tck/issues/31 to follow-up on this, though I know it's really supposed to be published correctly the first time.)

Finally I would probably choose to squash some of these commits.. maybe even to a single commit.  That's just a preference; if you want to leave it as 16 commits that's just as valid.   Will leave it up to you.

**Review by scottkurz — APPROVED** [review]

**Pull request merged**
